### ISSUE TITLE
Get market's order book API improvement proposal

### ISSUE THREAD

**Issue by luzzif** [Loopring/DEX-API#44]
A proposal to present an improved version of the get market's order book API. An adapted version of its OpenAPI (written in YAML) definition can be found [here](https://github.com/Loopring/DEX-API/files/5467829/swagger.txt).

A question that can help further improve the API:
What's the difference between a given bid/ask size and volume in the current API?

**Comment by yueawang** [comment]
> What's the difference between a given bid/ask size and volume in the current API?

Volume is given in base token, and size for quota token, for example, in the LRC-USDT market, one order sells 100LRC at 0.2, so the order volume is 100, then the size is 20. Relay gives this to prevent UI from potential precision errors.

**Comment by luzzif** [comment]
> > What's the difference between a given bid/ask size and volume in the current API?
> 
> Volume is given in base token, and size for quota token, for example, in the LRC-USDT market, one order sells 100LRC at 0.2, so the order volume is 100, then the size is 20. Relay gives this to prevent UI from potential precision errors.

Oh okok. Maybe we can rename these to make the meaning clearer, or we could just update the field's description I guess (I'll think about that and eventually attach the updated Swagger definition to the OP).
What I got from your description is also that the size is expressed in bips it seems like (0.2 USDT gives a size of 20)? Even though if I'm not mistaken, bips are generally used to refer to percentages and not absolute quantities. Was it a mistake from your side or have I just misunderstood?

**Comment by yueawang** [comment]
> What I got from your description is also that the size is expressed in bips it seems like (0.2 USDT gives a size of 20)? Even though if I'm not mistaken, bips are generally used to refer to percentages and not absolute quantities. Was it a mistake from your side or have I just misunderstood?

0.2 is the price, in that case, LRC volume is 100, so ETH size == 100 * 0.2 == 20, not a bips.

**Comment by luzzif** [comment]
Oh, my bad sorry! 
And why are size, volume and price specified all together? My guess would be that for an order book to be usable for market analysis for example, only the order's volume and price would be needed, no?

Example on market LRC-USDT

Buy order example:
Buy 10 LRC @ 0.2 USDT

Sell order example
Sell 23 LRC @ 0.2 USDT

Only volume and price are needed. Prices are always expressed in quote tokens, while the volume refers to the base token (this has to be made clear in the documentation, even though it's pretty logical).

**Comment by yueawang** [comment]
Yes, size is just an aux data for UI display. The origin proposal of size is to avoid precision differences exist in Relay and UI or other subscribers, but usually the difference too small to take care of, just in case users raise bugs for it.

**Comment by luzzif** [comment]
Okok, so I think if we can simplify the order book order model, devs would benefit from it.
Because this was one of the not-so-clear points I encountered in the API when developing the current Swap UI for example.
